src/Npgsql/ThrowHelper.cs

```diff
@@ -7,6 +8,12 @@ namespace Npgsql
 {
     static class ThrowHelper
     {
+        [DoesNotReturn]
+        internal static void ThrowInvalidCastException_NotSupportedType(NpgsqlTypeHandler handler, NpgsqlParameter? parameter, Type type) =>
+            throw new InvalidCastException(parameter is null
```

Comment:
We can also handle nameless parameters here (i.e. when invoking functions with positional style), better to output the first message I think...

Reply:
What did you mean in the last sentence?

Reply:
I mean that if a parameter exists (non-null), but its name isn't set, we will currently throw with the second message, `... for parameter ''`, which isn't very nice - better to just omit it.

Reply:
Ah, I see. Will fix this.

Reply:
If you really want to go all the way, IIRC the parameter has a reference to its collection, so you could find its index and say "for parameter #4".